libopae/plugins/xfpga/sysfs.c

```diff
@@ -1720,6 +1745,31 @@ struct _fpga_object *alloc_fpga_object(const char *sysfspath, const char *name)
 		obj->objects = NULL;
 	}
 	return obj;
+out_err:
+	if (obj) {
+		free(obj);
+		obj = NULL;
+	}
+	return obj;
+}
+
+fpga_result destroy_fpga_object(struct _fpga_object *obj)
+{
+	FREE_IF(obj->path);
+	FREE_IF(obj->name);
+	FREE_IF(obj->buffer);
+	while (obj->size && obj->objects) {
+		if (destroy_fpga_object((struct _fpga_object *)
+						obj->objects[--obj->size])) {
+			FPGA_ERR("Error freeing subobject");
```

Comment:
Capture the returned result here / propagate it upwards.

Reply:
I think we should go on to free stuff and just remember the failed result to return at the end.